lxd/device/device_utils_proxy.go

```diff
@@ -55,16 +55,19 @@ func ProxyParseAddr(data string) (*deviceConfig.ProxyAddress, error) {
 	// Split <ports> into individual ports and port ranges.
 	ports := strings.SplitN(port, ",", -1)
 
+	newProxyAddr.PortRanges = make([][2]uint64, len(ports))
 	newProxyAddr.Ports = make([]uint64, 0, len(ports))
 
-	for _, p := range ports {
+	for i, p := range ports {
 		portFirst, portRange, err := network.ParsePortRange(p)
 		if err != nil {
 			return nil, err
 		}
-
-		for i := int64(0); i < portRange; i++ {
-			newProxyAddr.Ports = append(newProxyAddr.Ports, uint64(portFirst+i))
+		
+		// Inclusive range for ports
```

Comment:
We tend to use full stops at end of comments now.

Reply:
Can we keep the `PortRanges` field as "startNumber" and then "rangeSize" rather than "endNumber", with a single port having a range size of 1? This would keep it aligned with what is returned from `ParsePortRange`, in fact you can probably use the returned values from `ParsePortRange` directly.